generate_network.py

```diff
@@ -53,26 +59,20 @@ def init_arg_parser() :
 #            default=1, dest='num_edges', 
 #            help='The number of edges in the Waku network', 
 #            type=int, metavar='#edges>')
-    return parser
+    return parser.parse_args()
 
-# https://networkx.org/documentation/stable/reference/generated/networkx.generators.degree_seq.configuration_model.html
-def generate_config_model(n):
-    #degrees = nx.random_powerlaw_tree_sequence(n, tries=10000)
-    degrees = [random.randint(1, n) for i in range(n)]
-    if (sum(degrees)) % 2 != 0:         # adjust the degree sum to be even
-        degrees[-1] += 1
-    G = nx.configuration_model(degrees) # generate the graph
-    return G
 
+# Generate a random string (UC chars) of len n
 def generate_topic_string(n):
     rs = ""
     for _ in range(n):
         r = random.randint(65, 65 + 26 - 1) # only letters
         rs += (chr(r))                      # append the char generated
     return rs
```

Comment:
Nitpick, you can aggregate this into a single string build:

```python
return "".join(chr(random.randint(65, 65 + 26 - 1)) for _ in range(n))
```

Reply:
I do agree comprehension is shorter, but I just wanted to split stuff so that I can add comment to individual steps. Improves the readability a bit.  Just to leaving it for the moment

Reply:
In that case you can abstract a generator that returns only letters. `random.choice(string.ascii_lowercase)` (iirc) could be a more explicit choice.
But it can always be changed later on if needed.

Reply:
that's a defo lot nicer to read, cheers! let me take a look at fix that
in the next update.